core/src/main/scala/kafka/log/Log.scala

```diff
@@ -678,14 +679,34 @@ class Log(@volatile var dir: File,
   }
 
   /**
-   * Rename the directory of the log
+   * Mark the topic partition log folder for async deletion by moving it under the
+   * 'delete' folder
+   */
+  def markForDeletion(): Unit = {
+    // create an empty file with the name of the corresponding topic partition
+    val tpFile = new File(dir, topicPartition.toString)
+    if (!tpFile.exists())
+      tpFile.createNewFile()
+    // create a 'delete' folder (if it does not exist already) to place all topic partitions
+    // marked for deletion
+    val deleteDir = new File(dir.getParentFile, DeleteDirParent)
+    if (!deleteDir.exists || !deleteDir.isDirectory)
+      deleteDir.mkdirs()
```

Comment:
If !deleteDir.isDirectory is true, would this call work ?

Reply:
Yes, this call is made if there is no file/folder with than name, or if there is a non-directory item (file) with that name. So the call would create the folder. If a directory with this name already exists the call would be skipped.

Were you thinking of something specific that could break this?

Reply:
Here is the beginning of mkdirs:
```
    public boolean mkdirs() {
        if (exists()) {
            return false;
```
If the deleteDir.isDirectory evaluates to false, mkdirs call is a no-op.
I don't think this is what you wanted.